addons/sale/static/src/js/variant_mixin.js

```diff
@@ -565,6 +602,20 @@ var VariantMixin = {
      */
     _toggleDisable: function ($parent, isCombinationPossible) {
         $parent.toggleClass('css_not_available', !isCombinationPossible);
+        if ($parent.hasClass('in_cart')) {
+            let $modalContent = $parent.parents('.modal-content');
+            let $footer = $modalContent.find('.modal-footer');
+            let $calledFromShop = $modalContent.find('.oe_optional_products_modal.oe_website_sale');
+            let primaryButton = $footer.find('.btn-primary');
+            primaryButton.prop('disabled', !isCombinationPossible);
+            primaryButton.toggleClass('disabled', !isCombinationPossible);
+
+            if ($calledFromShop.length) {
+                let secondaryButton = $footer.find('.btn-secondary');
+                secondaryButton.prop('disabled', !isCombinationPossible);
+                secondaryButton.toggleClass('disabled', !isCombinationPossible);
+            }
+        }
```

Comment:
You don't have to agree but don't you think it is easier to read when written like this?
Less use of variable, no duplicate line
```javascript
if ($parent.hasClass('in_cart')) {
    const $modalContent = $parent.parents('.modal-content');
    const btnToHandle = ['.btn-primary'];
    if ($modalContent.find('.oe_optional_products_modal.oe_website_sale').length) {
        // If called from shop
        btnToHandle.push('.btn-secondary');
    }
    $modalContent.find('.modal-footer').find(btnToHandle.join(','))
                 .prop('disabled', !isCombinationPossible)
                 .toggleClass('disabled', !isCombinationPossible);
}
```

`oe_website_sale` mentioned inside the `sale` module.. :cry: but don't worry, I saw it was already mentionned before your PR so.. keep it like that :shrug:

Reply:
I've moved it to the website_sale variant_mixin.js and override it with the website_sale specific logic. Also reduced the number of variables :D